# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution
</div>
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import math

In [3]:
data = pd.io.stata.read_stata('us_job_market_discrimination.dta')

FileNotFoundError: [Errno 2] No such file or directory: 'us_job_market_discrimination.dta'

In [ ]:
# number of callbacks for black-sounding names
sum(data[data.race=='w'].call)

In [ ]:
data.head()

<div class="span5 alert alert-success">
<p>Your answers to Q1 and Q2 here</p>
</div>

Chi-Square test is applicable as this is a Categorical Data.We can use the chi-square test for independence.As the sample size is greater than 30 obervations we can apply Central Limit theorem to this Data and assume that all the data points are independent to each other.

# 2.Hypothesis

H0- People with black sounding name have equal chance of being called back.


H1 - People with black sounding name have less chance of being called back

In [36]:
x=data[['race','call']]

# Answer to Q3

Lets Assume 95% confidence interval

In [40]:
num_black=len(x[x.race=="b"])
num_white=len(x[x.race=="w"])
print(n1,n2)

2435 2435


In [41]:
callback_black=sum(x[x.race=="b"]['call'])
callback_black

callback_white=sum(x[x.race=="w"]['call'])
print(callback_black,callback_white)

157.0 235.0


In [42]:
percent_white = callback_white/num_white
percent_white

0.09650924024640657

In [43]:
percent_black = callback_black/num_black
percent_black

0.06447638603696099

In [44]:
diff=percent_white-percent_black


In [45]:
Std_error_black = (percent_black * (1 - percent_black)) / num_black
Std_error_white = (percent_white * (1 - percent_white)) / num_white
SE = math.sqrt(Std_error_black + Std_error_white)
SE

0.0077833705866767544

<div class="span5 alert alert-success">
<p> Your answers to Q4 and Q5 here </p>
</div>

In [53]:
#For 95% confidence interval, the critical z value is 1.96.

ME= 1.96*SE
ME

0.015255406349886438

In [54]:
CI=[diff-ME, diff+ME]
CI

[0.016777447859559147, 0.047288260559332024]

In [55]:
Z = (diff-0)/SE
Z

4.11555043573

Using Scipy.stats

In [57]:
import scipy.stats as stats
stats.ttest_ind(x[x.race=="b"]['call'], x[x.race=="w"]['call'],equal_var=False)

Ttest_indResult(statistic=-4.114705290861751, pvalue=3.942941513645935e-05)

We can say that as our z statistics is 4.11 for using bootstrap and stats package and p values is less than 0.05 so we
can reject null hypothesis.We can say that there is a significant difference between the callback rates for black sounding 
and white names

I dont think that race/name is the most important factor in callback success.It may need to find out the relation with respect
to other factors about education and work experience etc before making such assumption

In [ ]:
pwd()